### PR TITLE
Moment Banner

### PR DESCRIPTION
**Pull request by jlieb10**
## What does this change?
New banner for upcoming moment

TBD: Pre-launch style tweaks, update campaign link

## Screenshots
Non-supporters - Desktop wide
banner_nonsupporters_desktop

Non-supporters - Desktop small
banner_nonsupporters_desktop_small

Non-supporters - Tablet
banner_nonsupporters_tablet

Non-supporters - Mobile
banner_nonsupporters_mobile

Supporters - Desktop wide
banner_supporters_desktop

Supporters - Desktop small
banner_supporters_desktop_small

Supporters - Tablet
banner_supporters_tablet

Supporters - Mobile
banner_supporters_mobile

## What is the value of this and can you measure success?
The banner will be one of the tools we use to measure the success of the moment using AV of contributions

### Accessibility test checklist

_Note: The banner and banners in general do not work well with screen readers. We have not adopted a best practice for solving this problem at this point_
- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [X] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [X] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Review by ionamckendrick — APPROVED** [review]
Looks wonderful! 👏 
As discussed, we can revisit some line heights / column widths nearer launch (to avoid ragged edges like on small desktop)

File: common/app/conf/switches/ABTestSwitches.scala [hunk 1]
```diff
@@ -66,6 +66,26 @@ trait ABTestSwitches {
     exposeClientSide = true
   )
 
+  Switch(
+    ABTests,
+    "ab-contributions-banner-environment-moment-supporters",
+    "new moment lorem ipsum",
```
Comment (tjsilver) [inline common/app/conf/switches/ABTestSwitches.scala]: will the 'lorem ipsum' text be removed before go-live?

Comment (jlieb10) [reply common/app/conf/switches/ABTestSwitches.scala]: 👍 fixed

File: static/src/javascripts/projects/common/modules/commercial/templates/acquisitions-banner-moment.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            </div>
+            
+            <div class="moment-banner__copy">
+                <!-- Render bold lead sentence if it exists -->
+                ${
+                    params.leadSentence
+                        ? `<span class="moment-banner__copy-in-bold">${
+                              params.leadSentence
+                          }</span>`
+                        : ''
```
Comment (tjsilver) [inline static/src/javascripts/projects/common/modules/commercial/templates/acquisitions-banner-moment.js]: Could you write `params.leadSentence && `<span class="moment-banner__copy-in...` to save using the ternary?

Comment (jlieb10) [reply static/src/javascripts/projects/common/modules/commercial/templates/acquisitions-banner-moment.js]: no sadly that's react only -- doing it like this here renders an 'undefined' string - but good thinking!

File: static/src/javascripts/projects/common/modules/commercial/templates/acquisitions-banner-moment.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                    <a tabindex="3" class="button  engagement-banner__button  engagement-banner__button--moment-link" href="${
+                        params.linkUrl
+                    }">
+                        Read our pledge
+                    </a>
+                </div>
+                
+                <div class="engagement-banner__cta">
+                    <a tabindex="3" class="button engagement-banner__button" href="${
+                        params.linkUrl
```
Comment (tjsilver) [inline static/src/javascripts/projects/common/modules/commercial/templates/acquisitions-banner-moment.js]: Pledge and CTA links are the same - we discussed that the pledge link will be changed when it's available.

**Review by tomrf1 — APPROVED** [review]
brilliant!

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-moment-banner-environment.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+const defaultMobileCopy =
+    'This is The Guardian’s pledge: we will be truthful, resolute and undeterred in pursuing our journalism on the environment. Support from our readers makes this work possible.';
+
+const thankYouCopy =
+    'The climate emergency is the defining issue of our times. This is The Guardian’s pledge: we will be truthful, resolute and undeterred in pursuing our journalism on the environment. We will give global heating, wildlife extinction and pollution the urgent attention they demand. Our independence means we can interrogate inaction by those in power. ';
+const thankYouMobileCopy =
+    'The climate emergency is the defining issue of our times. This is The Guardian’s pledge: we will be truthful, resolute and undeterred in pursuing our journalism on the environment. ';
```
Comment (tjsilver) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-moment-banner-environment.js]: Probably not important - but there's a stray space after the full stop on this one.

**Review by tjsilver — COMMENTED** [review]
great work!

**Pull request merged**